### PR TITLE
init-model

### PR DESCRIPTION
**Pull request by phamhuycuong**

### CODE REVIEW

**Comment by phamkykhoi** [comment]
Okie.

File: app/models/category.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class Category < ActiveRecord::Base
+  has_many :word, dependent: :destroy
```
Comment (dieunb) [inline app/models/category.rb]: `has_many` have to plural `words`

File: app/models/lesson.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Lesson < ActiveRecord::Base
+  has_many :word
```
Comment (dieunb) [inline app/models/lesson.rb]: as above comment

File: app/models/lesson.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+  has_many :lesson_word, dependent: :destroy
```
Comment (dieunb) [inline app/models/lesson.rb]: as above comment

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+class User < ActiveRecord::Base
+  has_many :lesson, dependent: :destroy
```
Comment (dieunb) [inline app/models/user.rb]: as above comment

File: app/models/word.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class Word < ActiveRecord::Base
+  has_many :lesson_word, dependent: :destroy
```
Comment (dieunb) [inline app/models/word.rb]: as above comment

File: app/models/answer.rb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+class Answer < ActiveRecord::Base
+  belongs_to :words
```
Comment (dieunb) [inline app/models/answer.rb]: `word` not `words`

File: app/models/lesson.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+class Lesson < ActiveRecord::Base
+  has_many :words
```
Comment (dieunb) [inline app/models/lesson.rb]: I think you want to add `through: :lesson_words` at here to define relation `n-m` between `lesson` and `word` model

File: app/models/word.rb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+class Word < ActiveRecord::Base
+  has_many :lesson_words, dependent: :destroy
+  has_many :answers, dependent: :destroy
```
Comment (dieunb) [inline app/models/word.rb]: Missing relationship with `lesson` model. Notice above comment

**Comment by phamkykhoi** [comment]
Ok

File: app/models/lesson.rb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+class Lesson < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :category
+
+  has_many :words
+  has_many :lesson_words, through: :lesson_words, dependent: :destroy
```
Comment (dieunb) [inline app/models/lesson.rb]: Ở đây bị ngược anh nhé, `through` phải đặt ở trên `has_many :words`

File: app/models/word.rb [hunk 10]
```diff
@@ -0,0 +1,7 @@
+class Word < ActiveRecord::Base
+  belongs_to :category
+  belongs_to :lesson
+
+  has_many :lesson_words, through: :lesson_words, dependent: :destroy
```
Comment (dieunb) [inline app/models/word.rb]: Đây cũng vậy ạ

**Comment by dieunb** [comment]
OK

**Pull request merged**